dtos/device.go

```diff
@@ -35,18 +36,19 @@ type Device struct {
 // UpdateDevice and its properties are defined in the APIv2 specification:
 // https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-metadata/2.1.0#/UpdateDevice
 type UpdateDevice struct {
-	Id             *string     `json:"id" validate:"required_without=Name,edgex-dto-uuid"`
-	Name           *string     `json:"name" validate:"required_without=Id,edgex-dto-none-empty-string,edgex-dto-rfc3986-unreserved-chars"`
-	Description    *string     `json:"description" validate:"omitempty"`
-	AdminState     *string     `json:"adminState" validate:"omitempty,oneof='LOCKED' 'UNLOCKED'"`
-	OperatingState *string     `json:"operatingState" validate:"omitempty,oneof='UP' 'DOWN' 'UNKNOWN'"`
-	LastConnected  *int64      `json:"lastConnected"`
-	LastReported   *int64      `json:"lastReported"`
-	ServiceName    *string     `json:"serviceName" validate:"omitempty,edgex-dto-none-empty-string,edgex-dto-rfc3986-unreserved-chars"`
-	ProfileName    *string     `json:"profileName" validate:"omitempty,edgex-dto-none-empty-string,edgex-dto-rfc3986-unreserved-chars"`
-	Labels         []string    `json:"labels"`
-	Location       interface{} `json:"location"`
-	AutoEvents     []AutoEvent `json:"autoEvents" validate:"dive"`
+	Id             *string                `json:"id" validate:"required_without=Name,edgex-dto-uuid"`
+	Name           *string                `json:"name" validate:"required_without=Id,edgex-dto-none-empty-string,edgex-dto-rfc3986-unreserved-chars"`
+	Description    *string                `json:"description" validate:"omitempty"`
+	AdminState     *string                `json:"adminState" validate:"omitempty,oneof='LOCKED' 'UNLOCKED'"`
+	OperatingState *string                `json:"operatingState" validate:"omitempty,oneof='UP' 'DOWN' 'UNKNOWN'"`
+	LastConnected  *int64                 `json:"lastConnected"`
+	LastReported   *int64                 `json:"lastReported"`
+	ServiceName    *string                `json:"serviceName" validate:"omitempty,edgex-dto-none-empty-string,edgex-dto-rfc3986-unreserved-chars"`
+	ProfileName    *string                `json:"profileName" validate:"omitempty,edgex-dto-none-empty-string,edgex-dto-rfc3986-unreserved-chars"`
+	Labels         []string               `json:"labels"`
+	Location       interface{}            `json:"location"`
+	Tags           map[string]interface{} `json:"tags,omitempty"`
```

Comment:
in the UpdateXXX DTO fields, we don't add omitempty, because we have to differentiate the default empty value (empty slice []) and nil.

Reply:
Done